src/rules/linalg/factorization.jl

```diff
@@ -12,14 +12,17 @@ end
 
 function rrule(::typeof(getproperty), F::SVD, x::Symbol)
     if x === :U
-        return F.U, (Rule(Ȳ->(U=Ȳ, S=zero(F.S), V=zero(F.V))), DNERule())
+        rule = Ȳ->(U=Ȳ, S=zero(F.S), V=zero(F.V))
```

Comment:
For efficiency's sake, should we be using `Zero()` here, as opposed to `zero(F.S)` and `zero(F.V)`? (Or better still, `Zeros` from `FillArrays`, since that retains shape information)

Reply:
Yeah I thought about `Zero()` but decided not to pursue it for now because it loses size information. I wonder if we could be a bit fancy with the definition of `Zero` such that it retains information when applicable as type parameters.

The issue with using FillArrays, which I ran into when exploring their use in Nabla for a similar purpose, is that they're `AbstractArray`s for which `setindex!` is not defined, which means a lot of internal code has to be audited in order to not make assumptions about being able to assign indices for arbitrary array inputs.

Reply:
Hmm okay, fair enough. I do think that we should go down the `FillArrays` route at some point. I'll raise an issue about this so that we don't forget, and revisit later.

I don't think I'm pro encoding size information as a type parameter as I imagine it would play badly with the compiler if we had a lot of differently-sized `Zero` things, which I could well imagine happening.

Reply:
I'm only against it insofar as I'm not sure that it's a significant enough gain to be worth the effort. Can you elaborate on how it would play poorly with the compiler? If we have size information as type parameters, say as a tuple, we could allow e.g. `Zero{(2,3)}() * Zero{(3,4)}` which would produce a `Zero{(2,4)}`.

Reply:
Sure. My concern with the compiler is that a proliferation of types (which, as I understand it, is what having loads of instances of parametric types essentially does) will increase compile times substantially, because you have lots of types. I'm far from expert here though, so I could be completely wrong.

The other thing is that I doubt that we really don't lose much performance by using dynamic sizing. i.e. `Zeros(2, 3) * Zeros(3, 4)` is still _really_ fast.

Reply:
Appropriate `@nospecialize` solves

Reply:
Sure. But then we might as well just use `FillArrays` and be done with it, since it's already defined all of these things and is something I think we should be supporting anyway.

Reply:
This is a bit out of scope for the current PR so we should continue this discussion in an issue.